scripts/monitoring/auto_scale_workers.py

```diff
@@ -0,0 +1,256 @@
+import os
+import re
+import time
+import uuid
+import pytz
+import requests
+
+import boto3
+
+from django.conf import settings
+import botocore
+
+# from botocore.exceptions import ClientError
+# from http import HTTPStatus
+
+utc = pytz.UTC
+
+# TODO: Check if need some kind of logging/logger here.
+#
+# TO CODE
+# Fetch all challenges from EvalAI
+# BOTO3 to get number of pending submissions in the queue.
+# SQS to set the number of workers to 1 or 0.
+
+# Eval AI related credentials
+evalai_endpoint = os.environ.get("API_HOST_URL")
+authorization_header = {
+    "Authorization": "Bearer {}".format(os.environ.get("AUTH_TOKEN"))
+}
+
+
+def execute_get_request(url):
+    response = requests.get(url, headers=authorization_header)
+    return response.json()
+
+
+def get_challenges():
+    all_challenge_endpoint = (
+        "{}/api/challenges/challenge/all/unapproved/all".format(
+            evalai_endpoint  # Gets all unapproved challenges
+        )
+    )
+    response = execute_get_request(all_challenge_endpoint)
+
+    return response.json()
+
+
+# TODO: Check if we can use challenge["queue"] directly to get queue name.
+# TODO: Understand if all the attributes are exposed in the API calls, or are they done selectively.
+def get_sqs_queue_by_name(queue_name):
+    """
+    Returns:
+        Returns the SQS Queue object
+    """
+    if settings.DEBUG or settings.TEST:
```

Comment:
we don't need to handle this case. I think these checks are from our `aws_utils.py` method. When we will run the script it will be running independent of django server. So this method can just fetch the boto3 client for sqs.

Reply:
Fixed.